test/distributed/org/apache/cassandra/fuzz/sai/StaticsTortureTest.java

```diff
@@ -61,206 +52,145 @@ public void staticsTortureTest()
                                        schema.staticColumns.get(0).name));
-        sut.schemaChange(String.format("CREATE INDEX %s_%s_sai_idx ON %s.%s (%s) USING 'sai';",
+        cluster.schemaChange(String.format("CREATE INDEX %s_%s_sai_idx ON %s.%s (%s) USING 'sai';",
                                        schema.table,
                                        schema.staticColumns.get(1).name,
                                        schema.keyspace,
                                        schema.table,
                                        schema.staticColumns.get(1).name));
-        sut.schemaChange(String.format("CREATE INDEX %s_%s_sai_idx ON %s.%s (%s) USING 'sai';",
+        cluster.schemaChange(String.format("CREATE INDEX %s_%s_sai_idx ON %s.%s (%s) USING 'sai';",
                                        schema.table,
                                        schema.staticColumns.get(2).name,
                                        schema.keyspace,
                                        schema.table,
                                        schema.staticColumns.get(2).name));
-        DataTracker tracker = new DefaultDataTracker();
+
         TokenPlacementModel.ReplicationFactor rf = new TokenPlacementModel.SimpleReplicationFactor(cluster.size());
-        ReplayingHistoryBuilder history = new ReplayingHistoryBuilder(SEED + idx,
-                                                                      MAX_PARTITION_SIZE,
-                                                                      100,
-                                                                      tracker,
-                                                                      new DoubleWritingSut(sut,
-                                                                                           new QueryModifyingSut(sut,
-                                                                                                                 schema.keyspace + "." + schema.table,
-                                                                                                                 debugSchema.keyspace + "." + debugSchema.table)),
-                                                                      schema,
-                                                                      rf,
-                                                                      SystemUnderTest.ConsistencyLevel.QUORUM);
 
+        Generator<BitSet> regularColumnBitSet = Generators.bitSet(schema.regularColumns.size());
+        Generator<BitSet> staticColumnBitSet = Generators.bitSet(schema.staticColumns.size());
         EntropySource rng = new JdkRandomEntropySource(1l);
+
+        ReplayingHistoryBuilder history = new ReplayingHistoryBuilder(schema.valueGenerators, hb -> {
+            return InJvmDTestVisitExecutor.builder().pageSizeSelector(i -> rng.nextInt(1, 10)).build(schema, hb, cluster);
+        });
+
+
         long[] values = new long[UNIQUE_CELL_VALUES];
```

Comment:
I guess we don't need the `values` array now...just `UNIQUE_CELL_VALUES`?

Reply:
Yes, it is now redundant, thank you for spotting this. I over-focused on porting the test, so missed this.